jetstream/sponsored-suggestion-placement-phase-2.toml

```diff
@@ -0,0 +1,694 @@
+    COUNTIF(exposed_session AND new_session_start)
+)"""
+data_source = "urlbar_search_sessions"
+friendly_name = "Ad clicks per exposed search session"
+description = "Per-client rate of SERP ad clicks per exposed urlbar search session"
+
+[metrics.exposed_ad_clicks_per_session.statistics.bootstrap_mean]
+[metrics.exposed_ad_clicks_per_session.statistics.deciles]
+
+[metrics.exposed_adm_impressions_per_session]
+select_expression = """SAFE_DIVIDE(
+    COUNTIF(exposed_session AND urlbar_fields.has_sponsored_impression),
+    COUNTIF(exposed_session AND new_session_start)
+)"""
+data_source = "urlbar_search_sessions"
+friendly_name = "Sponsored suggestion impressions per exposed search session"
+description = "Per-client rate of sponsored suggestion impressions per exposed urlbar search session"
+
+[metrics.exposed_adm_impressions_per_session.statistics.bootstrap_mean]
+[metrics.exposed_adm_impressions_per_session.statistics.deciles]
+
+[metrics.exposed_adm_clicks_per_session]
+select_expression = """SAFE_DIVIDE(
+    COUNTIF(
+        exposed_session
+        AND urlbar_fields.event_action = 'engaged'
+        AND urlbar_fields.product_selected_result = 'admarketplace_sponsored'
+    ),
+    COUNTIF(exposed_session AND new_session_start)
+)"""
+data_source = "urlbar_search_sessions"
+friendly_name = "Sponsored suggestion clicks per exposed search session"
+description = "Per-client rate of sponsored suggestion clicks per exposed urlbar search session"
+
+[metrics.exposed_adm_clicks_per_session.statistics.bootstrap_mean]
+[metrics.exposed_adm_clicks_per_session.statistics.deciles]
+
+[data_sources]
+
+[data_sources.tagged_google_serp_events]
+from_expression = """(
+    SELECT
+        * EXCEPT (is_ad_component),
+        component IN ('ad_carousel', 'ad_image_row', 'ad_link', 'ad_sidebar',
+        'ad_sitelink') AS ad_component
+    FROM `mozdata.firefox_desktop.serp_events`
+    WHERE search_engine = 'google' AND is_tagged
+)"""
+client_id_column = "legacy_telemetry_client_id"
+experiments_column_type = "native"
+submission_date_column = "submission_date"
+friendly_name = "Glean Google tagged SERP events"
+description = "Glean SERP events aggregated by impression, component for tagged Google search"
+
+[data_sources.urlbar_search_sessions]
+friendly_name = "Urlbar search sessions"
+description = "Search sessions initiated in the urlbar including urlbar and tagged Google serp Glean events"
+client_id_column = "legacy_telemetry_client_id"
+submission_date_column = "submission_date"
+experiments_column_type = "native"
+from_expression = """(
+    WITH exposures AS (
+      -- urlbar.exposure events sequence info
+      SELECT
+        client_info.client_id AS glean_client_id,
+        ping_info.seq,
+        event_timestamp,
+        TRUE AS exposed,
+      FROM
+        `mozdata.firefox_desktop.events_unnested`
+      WHERE
+        event_category = 'urlbar' AND event_name = 'exposure'
```

Comment:
We don't need to deal with exposures in this experiment, since all users that were "exposed" by typing a sponsored keyword actually saw a sponsored result, which is recorded in the urlbar events. Are stakeholders asking for an exposure-based analysis? If so, you can simplify these `urlbar*` CTEs to just adding a flag when there was a sponsored result impression.

Reply:
Hey, @dzeber . Agreed, but I'm staying consistent with the experiment brief, which does indicate exposure-based analysis. I'm happy to update this query as much as possible--so far I just copy-pasted your initial query since stakeholders are interested in the exact same set of metrics.

Reply:
Sounds good - so yeah, rather than querying for `urlbar.exposure` events, you can change `urlbar_joined` to
```sql
select
  *,
  (select logical_or(r.product_result_type = 'admarketplace_sponsored') from unnest(results) as r) as exposed
from ...
```

Reply:
Thanks, @dzeber . I adopted your tip and simplified the workflow. Overall resulting counts are identical.